File: C++/Hard/WrongAnswer/abbreviating-the-product-of-a-range_2.cpp
class Solution {
public:
    string abbreviateProduct(int left, int right) {
        
        int product = 1;
        for (int i = left; i <= right; ++i) {
            product *= i;
        }
        int count = 0;
        int temp = product;
        while (temp % 10 == 0) {
            ++count;
            temp /= 10;
        }
        int d = 0;
        temp = product;
        while (temp > 0) {
            ++d;
            temp /= 10;
        }
        if (d <= 10) {
            return to_string(product) + "e" + to_string(count);
        }
        else {
            return to_string(product / pow(10, d - 5)) + "..." + to_string(product % (int)pow(10, 5)) + "e" + to_string(count);
        }
    }
};

File: Rust/Hard/WrongAnswer/abbreviating-the-product-of-a-range_2.rs
impl Solution {
    pub fn abbreviate_product(left: i32, right: i32) -> String {
        
        let mut product = 1;
        let mut trailing_zeros = 0;
        for i in left..=right {
            product *= i;
            while product % 10 == 0 {
                product /= 10;
                trailing_zeros += 1;
            }
            product %= 100_000;
        }
        let mut result = product.to_string();
        if trailing_zeros > 0 {
            result += &format!("e{}", trailing_zeros);
        }
        result
    }
}